File: C++/Problems/Problems/problem_3774.cpp
#include "problem_3774.h"

#include <algorithm>

bool problem_3774::test() {
    std::vector nums { 5, 2, 2, 4 };
    int k = 2;

    int expected = 5;

    auto result = absDifference(nums, k);

    std::cout << result << std::endl;

    return expected == result;
}

int problem_3774::absDifference(std::vector<int> &nums, int k) {
    int n = nums.size();
    int result = 0;
    std::sort(nums.begin(), nums.end());
    for (int i = 0; i < k; ++i)
    {
        result -= nums[i];
        result += nums[n - i - 1];
    }

    return result;
}

File: Python/Problems/Problems/problem3774.py
import heapq
import math
import sys

from utils import print1DArray, print2DArray
from typing import List, Set
from collections import Counter

from .problem import Problem


class Problem3774(Problem):
    def test(self) -> bool:
        nums = [5, 2, 2, 4]
        k = 2

        expected = 5

        result = self.absDifference(nums, k)

        print(result)

        return expected == result

    def absDifference(self, nums: List[int], k: int) -> int:
        n = len(nums)
        result = 0
        nums.sort()
        for i in range(k):
            result -= nums[i]
            result += nums[n - i - 1]

        return result